### PR TITLE
Trim leading spaces from command

### PR DESCRIPTION
**Pull request by xlzior**

### CODE REVIEW

File: src/test/java/seedu/address/commons/util/StringUtilTest.java [hunk 1]
```diff
@@ -219,4 +220,26 @@ public void getDetails_nullGiven_throwsNullPointerException() {
...
+    public void removeExtraWhitespace_nullGiven_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> StringUtil.removeExtraWhitespace(null));
+    }
+
+    @Test
+    public void removeExtraWhitespace_validInputs_correctResult() {
+        assertEquals(StringUtil.removeExtraWhitespace(""), "");
+        assertEquals(StringUtil.removeExtraWhitespace("words words more words"), "words words more words");
+        assertEquals(StringUtil.removeExtraWhitespace("words    more    words       -flag"), "words more words -flag");
+        assertEquals(StringUtil.removeExtraWhitespace("words \t \n words"), "words words");
```
Comment (xlzior) [inline src/test/java/seedu/address/commons/util/StringUtilTest.java:243]: I removed extra whitespace characters in order to accommodate `ls(lots of spaces)-contacts`. all commands will now have extra whitespace characters removed, such that two words are always separated by a single space character

not sure if it's going to create problems

File: src/main/java/seedu/address/commons/util/StringUtil.java [hunk 2]
```diff
@@ -88,4 +88,16 @@ public static boolean isNonZeroUnsignedInteger(String s) {
...
+
+    /**
+     * Removes extra whitespace characters found within the command
+     * @param s input string
+     * @return string with extra whitespace characters replaced with a single space
+     */
+    public static String removeExtraWhitespace(String s) {
+        requireNonNull(s);
+        String[] result = Arrays.stream(s.split("\\s+"))
+                .toArray(String[]::new);
```
Comment (jiarong15) [inline src/main/java/seedu/address/commons/util/StringUtil.java:100]: Isn't s.split("\\s+") the same as code in line 99 and line 10

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java [hunk 3]
```diff
@@ -43,7 +44,7 @@
      * @throws ParseException if the user input does not conform the expected format
      */
     public Command parseCommand(String userInput) throws ParseException {
-        String trimmedUserInput = userInput.trim();
+        String trimmedUserInput = StringUtil.removeExtraWhitespace(userInput.trim());
```
Comment (jiarong15) [inline src/main/java/seedu/address/logic/parser/AddressBookParser.java:47]: suggest maybe to use the trim function in the removeExtraWhitespace() function? So everything happens there

**Review by sebbycake — APPROVED** [review]
LGTM

**Pull request merged**